test/windows/WSLATests.cpp

```diff
@@ -1674,6 +1655,198 @@ class WSLATests
+    {
+        const winrt::Windows::Web::Http::Filters::HttpBaseProtocolFilter filter;
+        filter.CacheControl().WriteBehavior(winrt::Windows::Web::Http::Filters::HttpCacheWriteBehavior::NoCache);
+
+        const winrt::Windows::Web::Http::HttpClient client(filter);
+
+        try
+        {
+            auto response = client.GetAsync(winrt::Windows::Foundation::Uri(Url)).get();
+            auto content = response.Content().ReadAsStringAsync().get();
+
+            if (expectedCode.has_value())
+            {
+                VERIFY_ARE_EQUAL(static_cast<int>(response.StatusCode()), expectedCode.value());
+            }
+            else
+            {
+                LogError("Unexpected reply for: %ls", Url);
+                VERIFY_FAIL();
+            }
+        }
+        catch (...)
+        {
+            auto result = wil::ResultFromCaughtException();
+
+            if (!expectedCode.has_value())
+            {
+                // We currently reset the connection if connect() fails inside the VM. Consider failing the Windows connect() instead.
+                VERIFY_ARE_EQUAL(result, HRESULT_FROM_WIN32(WININET_E_INVALID_SERVER_RESPONSE));
+            }
+            else
+            {
+                LogError("Expected success but request failed with 0x%08X for: %ls", result, Url);
+                VERIFY_FAIL();
+            }
+        }
+    }
+
+    void RunPortMappingsTest(WSLA_CONTAINER_NETWORK_TYPE Mode)
+    {
+        auto settings = GetDefaultSessionSettings();
+        settings.NetworkingMode = WSLANetworkingModeNAT;
+
+        auto session = CreateSession(settings);
+
+        // Test a simple port mapping.
+        {
+            WSLAContainerLauncher launcher(
+                "python:3.12-alpine", "test-ports", {}, {"python3", "-m", "http.server"}, {"PYTHONUNBUFFERED=1"}, Mode);
+
+            launcher.AddPort(1234, 8000, AF_INET);
+            launcher.AddPort(1234, 8000, AF_INET6);
+
+            auto container = launcher.Launch(*session);
+            auto initProcess = container.GetInitProcess();
+            auto stdoutHandle = initProcess.GetStdHandle(1);
+
+            // Wait for the container bind() to be completed.
+            WaitForOutput(stdoutHandle.get(), "Serving HTTP on 0.0.0.0 port 8000");
+
+            ExpectHttpResponse(L"http://127.0.0.1:1234", 200);
+            ExpectHttpResponse(L"http://[::1]:1234", {});
+
+            // Validate that the port cannot be reused while the container is running.
+            WSLAContainerLauncher subLauncher(
+                "python:3.12-alpine", "test-ports-2", {}, {"python3", "-m", "http.server"}, {"PYTHONUNBUFFERED=1"}, Mode);
+
+            subLauncher.AddPort(1234, 8000, AF_INET);
+            auto [hresult, newContainer] = subLauncher.LaunchNoThrow(*session);
+            VERIFY_ARE_EQUAL(hresult, HRESULT_FROM_WIN32(ERROR_ALREADY_EXISTS));
+
+            VERIFY_SUCCEEDED(container.Get().Stop(9, 0));
```

Comment:
Do we unmap the host port upon container delete? Don't think I saw that. We should unmap host port on Delete and verify that here